samples/DLLMethodCaller/FortuneTeller.cs

```diff
@@ -7,14 +7,22 @@ namespace DLLMethodCaller
 {
     class FortuneTeller
     {
+        private int total;
         public void AcceptMoney(int amount)
         {
-            // TODO
+            total += amount;
         }
 
         public string PredictFuture(DateTime birthday, bool isMarried)
         {
-            // TODO
+            if (isMarried)
+                return "You're broke!";
+
+            Random random = new Random(int.Parse(birthday.ToString("yyyyMMdd")));
+            lock (random) // synchronize
+            {
+                return $"You're rich, You're worth {random.Next(0, total)} today!";
+            }
```

Comment:
I propose some improvements to the logic of this method:
1. check if total == 0 and return something appropriate immediately :)
1. if next random is relatively small, maybe it is not wise to say "You're rich", so maybe change wording there.
1. at the end you should reset the total to 0, because fortune tellers don't work for free :)